docs/questions/visualizations/map.md

```diff
@@ -2,46 +2,195 @@
 ---
 
 # Maps
 
-When you select the **Map** visualization setting, Metabase will automatically try and pick the best kind of map to use based on the table or result set. Here are the maps that Metabase uses:
+Metabase has three types of map visualization:
+
+- [**Pin map**](#pin-maps) for putting individual data points on a map using longitude and latitude coordinates;
+- [**Grid map**](#grid-maps)
+- [**Region map**](#region-maps) for data broken out by regions, like countries or states. Metabase comes with two built-in maps, but you can upload your own custom regions.
+
+When you select the **Map** visualization setting, Metabase will automatically try and pick the best kind of map to use based on the table or result set, as long as the columns with the geographical data have the [right metadata](../../data-modeling/metadata-editing.md).
+
+![Map types](../images/map-types.png)
+
+{% include youtube.html id='EewOLxHht1s' %}
+
+## Pin map
+
+Pin maps display pins for individual data points on the map. They work best for displaying unaggregated geographical data.
+
+### Pin map data shape
+
+To build a pin map, you need a query that returns a result that has latitude and longitude columns. Metabase will put one pin on the map for each row in your table, based on the latitude and longitude fields. Other data in the rows will be shown in the tooltip, and won't otherwise affect the placement or shape of pins.
+
+![Pin map data shape](../images/pin-data-shape.png)
+![Pin map with a tooltip](../images/pin-map-with-tooltip.png)
+
+Note that the tooltips will only be displayed when the map is sufficiently zoomed in using [**Draw box to filter**](#draw-box-to-filter).
+
+### Create a pin map
+
+To create a pin map:
+
+1. Build a query with latitude and longitude columns for each data point (either in the query builder or using SQL);
+2. Select **Visualization**, and pick **Map**;
+3. If your query results have columns whose field type is set to latitude/longitude in [table metadata](../../data-modeling/metadata-editing.md#field-type), Metabase should build a pin map automatically.
+
+   Otherwise, click on the **Gear** icon to go to visualization settings, choose **Map type: Pin map**, and pick columns that contain latitude and longitude coordinates.
+
+### Pin maps display 2000 pins by default
+
+By default, pin maps will display 2000 pins even if there are more rows in your query results (this limit is the same for every chart displaying unaggregated data in Metabase). You can use the environment variable [`MB_UNAGGREGATED_QUERY_ROW_LIMIT`](../../configuring-metabase/environment-variables.md#mb_unaggregated_query_row_limit) to increase the number of data points rendered on charts based on unaggregated queries, but keep that this will affect all charts - not just the pin maps - and it might significantly slow down your Metabase and your browser.
```

Comment:
I added "(this limit is the same for every chart displaying unaggregated data in Metabase)" because people complain about 2000 on maps pins a lot but somehow don't complain about it on e.g. scatterplots...